### PR TITLE
fix: apps-view filter is no longer forgotten when returning from resources/k9s

### PR DESCRIPTION
**Pull request by darksworm**
## Summary

- After filtering the apps list, opening an app's resources, drilling into k9s and escaping back to apps, the filter was silently cleared. Restore it on the way back.
- `handleEscape` unconditionally wiped `UI.SearchQuery` / `UI.ActiveFilter` when leaving any non-apps view, and `SaveNavigationState` didn't capture the filter — so there was nothing to restore.
- Capture the filter onto `SavedNavigation` entries and reapply it in the `ViewTree → ViewApps` escape branch (also pops the matching saved entry to keep the stack bounded).

## Test plan

- [x] New e2e test `TestFilterPersistsAfterResourcesAndK9s` exercises the full filter → resources → k9s → esc round-trip. Verified red on `main`, green after fix.
- [x] Existing related e2e tests still pass (`TestK9s_*`, `TestTreeView*`, `TestContextSwitch*`, `TestSearchInputAcceptsAllCharacters`).
- [x] Full unit suite (`go test ./...`) green.
- [ ] Manual smoke: filter apps → Enter into resources → `K` into k9s → exit k9s → Esc — filter chip `<apps:…>` should still be visible.
- [ ] Manual smoke: filter apps → Esc once (no drill-down) — filter still clears on first Esc (existing edge case preserved).

## Summary by CodeRabbit

* **New Features**
  * Filters and search queries are now preserved when returning to the apps view from resource or tree views.

* **Tests**
  * Added end-to-end test verifying filter persistence after navigation and external actions.

[Review Change Stack](https://app.coderabbit.ai/change-stack/darksworm/argonaut/pull/242?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: e2e/filter_persistence_test.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+	cfgPath, err := tf.SetupWorkspace()
+	if err != nil {
+		t.Fatalf("setup workspace: %v", err)
+	}
+	if err := WriteArgoConfig(cfgPath, srv.URL); err != nil {
+		t.Fatalf("write config: %v", err)
+	}
+
+	mockK9s, _ := createMockK9s(t, tf.workspace, 0)
```
[the remark this answers is not in the document]
Comment (darksworm) [reply e2e/filter_persistence_test.go:39]: The finding is invalid: createMockK9s returns (scriptPath, argsFile string) — no error value. The existing call mockK9s, _ := createMockK9s(...) is discarding the unused argsFile (which other tests use to assert k9s args; we don't need that here), not an error. Same pattern is used in e2e/k9s_test.go:473 (TestK9s_TerminalRestoredAfterExit).
